homeassistant/components/sensor/mfi.py

```diff
@@ -0,0 +1,114 @@
+"""
+homeassistant.components.sensor.mfi
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Support for Ubiquiti mFi Sensors.
+
+Configuration:
+
+sensor:
+  platform: mfi
+  host: ADDRESS
+  port: PORT
+  username: USERNAME
+  password: PASSWORD
+
+Variable:
+
+host
+*Required
+ADDRESS is the IP or hostname of your mFi controller.
+
+port
+*Optional
+PORT is the port of your mFi controller (usually 6443)
+
+username
+*Required
+USERNAME is the mFi admin username
+
+password
+*Required
+PASSWORD is the mFi admin user's password
+"""
+
+import logging
+
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD
+from homeassistant.components.sensor import DOMAIN
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers import validate_config
+
+REQUIREMENTS = ['mficlient==0.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSOR_MODELS = [
+    'Ubiquiti mFi-THS',
+    'Ubiquiti mFi-CS',
+    'Outlet',
+]
+
+
+# pylint: disable=unused-variable
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up mFi sensors. """
+
+    if not validate_config({DOMAIN: config},
+                           {DOMAIN: ['host',
+                                     CONF_USERNAME,
+                                     CONF_PASSWORD]},
+                           _LOGGER):
+        _LOGGER.error('A host, username, and password are required')
+        return False
+
+    host = config.get('host')
+    port = int(config.get('port', 6443))
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+
+    from mficlient.client import MFiClient
+
+    try:
+        client = MFiClient(host, username, password, port=port)
+    except client.FailedToLogin as ex:
+        _LOGGER.error('Unable to connect to mFi: %s', str(ex))
+        return False
+
+    for device in client.get_devices():
+        for port in device.ports.values():
+            if port.model in SENSOR_MODELS:
+                add_devices([MfiSensor(port, hass)])
+
+
+class MfiSensor(Entity):
+    """ An mFi sensor that exposes tag=value. """
+
+    def __init__(self, port, hass):
+        self._port = port
+        self._hass = hass
+
+    @property
+    def name(self):
+        return self._port.label
+
+    @property
+    def state(self):
+        value = self._port.value
+        if self._port.tag == 'temperature':
+            temp_unit = self._hass.config.temperature_unit
+            if temp_unit.endswith('F'):
```

Comment:
There is no need for this. Home Assistant will [automatically convert temperature values](https://github.com/balloob/home-assistant/blob/dev/homeassistant/helpers/entity.py#L162-L169) to the user preferred temperature unit.